TestAttacherWithCSIDriver flakes

<!-- Please only use this template for submitting reports about failing tests in Kubernetes CI jobs -->

**Which jobs are failing**:

https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&text=TestAttacherWithCSIDriver

```
E0308 05:14:42.798363      15 csi_attacher.go:207] kubernetes.io/csi: attachment for vol02 failed: attacher error
E0308 05:14:54.451156      15 csi_attacher.go:188] kubernetes.io/csi: attacher.WaitForAttach timeout after 15s [volume=vol01; attachment.ID=csi-75f1d3164ac9ae6f73784b2ed6b77839daf5b01a8dd52e787c8769b6fdb1b8ec]
E0308 05:14:55.586363      15 csi_attacher.go:188] kubernetes.io/csi: attacher.WaitForAttach timeout after 15s [volume=vol02; attachment.ID=csi-8d666355799344c2b14819eb973950d28cf52f4d3a58e3fbeb8b92e29851fa17]
E0308 05:14:56.686716      15 csi_attacher.go:188] kubernetes.io/csi: attacher.WaitForAttach timeout after 15s [volume=vol02; attachment.ID=csi-fd8097578f4563d9a74d4e7ba8397c32e42b2cea4ae677f8ccb4ba12bbd0008f]
E0308 05:14:58.015137      15 csi_attacher.go:188] kubernetes.io/csi: attacher.WaitForAttach timeout after 15s [volume=test-vol; attachment.ID=csi-ee944121dabf8fe14295e9f2c7675639d989a9a6b2f5cee459cfbcd4b61e9b6b]
--- FAIL: TestAttacherWithCSIDriver (15.43s)
    --- FAIL: TestAttacherWithCSIDriver/CSIDriver_is_attachable (15.10s)
        csi_attacher_test.go:273: Attach() failed: attachment timeout for volume test-vol
        csi_attacher_test.go:276: Expected attachID, got nothing
```

/sig storage
/kind flake

--- a/pkg/volume/csi/csi_attacher_test.go
+++ b/pkg/volume/csi/csi_attacher_test.go
@@ -990,7 +990,7 @@ func newTestWatchPlugin(t *testing.T, fakeClient *fakeclient.Clientset) (*csiPlu
 		fakeClient = fakeclient.NewSimpleClientset()
 	}
 	fakeWatcher := watch.NewRaceFreeFake()
-	fakeClient.Fake.AddWatchReactor("*", core.DefaultWatchReactor(fakeWatcher, nil))
+	fakeClient.Fake.PrependWatchReactor("volumeattachments", core.DefaultWatchReactor(fakeWatcher, nil))
 	host := volumetest.NewFakeVolumeHostWithCSINodeName(
 		tmpDir,
 		fakeClient,

